pkg/scheduler/eventhandlers.go

```diff
@@ -246,16 +249,47 @@ func responsibleForPod(pod *v1.Pod, profiles profile.Map) bool {
 	return profiles.HandlesSchedulerName(pod.Spec.SchedulerName)
 }
 
+const (
+	// syncedPollPeriod controls how often you look at the status of your sync funcs
+	syncedPollPeriod = 100 * time.Millisecond
+)
+
+// WaitForHandlersSync waits for EventHandlers to sync.
+// It returns true if it was successful, false if the controller should shut down
+func (sched *Scheduler) WaitForHandlersSync(ctx context.Context) bool {
+	logger := klog.FromContext(ctx)
+	if err := wait.PollImmediateUntil(syncedPollPeriod,
+		func() (bool, error) {
+			for _, handler := range sched.registeredHandlers {
+				if !handler.HasSynced() {
+					return false, nil
+				}
+			}
+			return true, nil
+		},
+		ctx.Done()); err != nil {
+		logger.V(2).Error(err, "error occurred during WaitForHandlersSync")
```

Comment:
on a second thought: it makes more sense to return the error and let the caller log

And you can make the caller also log `Handlers synced`

Reply:
Updated. Also, in our use case, I think this error can only be a timeout error. Should we only log the message without `err`?
```
logger.V(2).Error(nil, "timeout during WaitForHandlersSync")
```